server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -308,26 +342,137 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
+    if (lookupConfig.getEnableLookupSyncOnStartup()) {
+      String tier = lookupListeningAnnouncerConfig.getLookupTier();
+      lookupBeanList = getLookupListFromCoordinator(tier);
+      if (lookupBeanList == null) {
+        LOG.info("Coordinator is unavailable. Loading saved snapshot instead");
+        lookupBeanList = getLookupListFromSnapshot();
+      }
+    } else {
+      lookupBeanList = getLookupListFromSnapshot();
+    }
+      if (lookupBeanList != null) {
+        ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
+        ListeningScheduledExecutorService executorService = MoreExecutors.listeningDecorator(
+            Executors.newScheduledThreadPool(
+                lookupConfig.getNumLookupLoadingThreads(),
+                Execs.makeThreadFactory("LookupReferencesManager-Startup-%s")
+            )
+        );
+        try {
+          List<ListenableFuture<Map.Entry>> futures = new ArrayList<>();
+          LOG.info("Starting lookup loading process");
+          for (LookupBean lookupBean : lookupBeanList) {
+            futures.add(
+                executorService.submit(
+                    () -> {
+
+                      LookupExtractorFactoryContainer container = lookupBean.getContainer();
+                      LOG.info(
+                          "Starting lookup [%s]:[%s]",
+                          lookupBean.getName(),
+                          container
+                      );
+                      if (container.getLookupExtractorFactory().start()) {
+                        LOG.info(
+                            "Started lookup [%s]:[%s]",
+                            lookupBean.getName(),
+                            container
+                        );
+                        return new AbstractMap.SimpleImmutableEntry<>(lookupBean.getName(), container);
+                      } else {
+                        LOG.error(
+                            "Failed to start lookup [%s]:[%s]",
+                            lookupBean.getName(),
+                            container
+                        );
+                        return null;
+                      }
+                    }
+                )
+            );
+          }
+          final ListenableFuture<List<Map.Entry>> futureList = Futures.allAsList(futures);
+          try {
+            futureList.get()
+                      .stream()
+                      .filter(Objects::nonNull)
+                      .forEach(builder::put);
+
+            stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
+
+            executorService.shutdownNow();
+            if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+              LOG.warn("Lookup loading didn't complete in 60 seconds!");
+            }
+          }
+          catch (InterruptedException e) {
+            LOG.error("Lookup loading interrupted. ");
```

Comment:
`futureList.cancel(true);` will not be called here, is it intended to make sure that is called?